app/models/user.rb

```diff
@@ -1,13 +1,22 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
+  before_destroy :at_least_one_admin?
   has_many :missions, dependent: :destroy
   has_secure_password
 
+  enum role: { user: 0, admin: 1 }
+
   # kaminali default mission in per page
   paginates_per 15
 
-  validates :name, presence: true
+  validates :name, :role, :password_confirmation, presence: true
   validates :email, presence: true, uniqueness: true
-  validates :password_confirmation, presence: true
+
+  private
+
+  def at_least_one_admin?
+    role = 'admin'
+    throw :abort if User.where(role: :admin).count == 1 && self.role == role
+  end
```

Comment:
命名不符合 Ruby 慣例：非回傳 boolean 的 method 名稱不應該有 `?` 結尾

Reply:
`self.role == role` 的部分看起來似乎不需要用到？再麻煩確認。
```suggestion
  def abort_if_last_admin
    throw :abort if User.where(role: :admin).size == 1
  end
```

Reply:
> `self.role == role` 的部分看起來似乎不需要用到？再麻煩確認。

如果無 `self.role `，會變成只要管理員只剩一人，"所有人"的資料皆不可刪除，而不是"不能刪除最後的管理員"

Reply:
那再麻煩將 `self.role == role` 的部分以 guard clause 呈現，除了更能夠表達意圖（ role 不等於 admin 時都可以刪除）外，也不用在每次刪除非 admin 時都還要執行 query `User.where(role: :admin).size` ：
```ruby
def abort_if_last_admin
  return unless role.admin?

  throw :abort if User.where(role: :admin).size == 1
end
```